spec/truffle/process/euid_spec.rb

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 2.0, or
+# GNU General Public License version 2, or
+# GNU Lesser General Public License version 2.1.
+
+require_relative '../../ruby/spec_helper'
+require 'etc'
+
+describe "Process.euid=" do
+
+  platform_is_not :windows do
+    as_user do
+      context "when ran by a non-root user" do
+        it "sets the effective user id to its own uid if given the username corresponding to its own uid" do
+          user = Etc.getpwuid(Process.uid).name
+          code = <<-RUBY
+            Process.euid = "#{user}"
+            puts Process.euid
+          RUBY
+
+          ruby_exe(code).should == "#{Process.uid}\n"
```

Comment:
Could we avoid the subprocess and just do it on the current process? Tests using subprocesses are much slower, and should only be used when there is no other way.

And let's an early check like `raise unless Process.uid == Process.euid` to make sure we don't actually change anything.